PowerNSX.psm1

```diff
@@ -21191,6 +21191,432 @@ function Remove-NsxSecurityGroupMember {
+        -entity $(Get-NsxSecurityGroup SG-PCI-Machines)
+
+    Statically specify the NSX Security Group called SG-PCI-Machines to be
+    included as part of the dynamic criteria
+
+    #>
+
+    [CmdletBinding(DefaultParameterSetName="search")]
+
+    param (
+        [Parameter(Mandatory=$true, ParameterSetName="search")]
+            [ ValidateSet("VM.NAME", "VM.GUEST_HOST_NAME", "VM.GUEST_OS_FULL_NAME", "VM.SECURITY_TAG", IgnoreCase=$true) ]
+            [String]$key,
+        [Parameter(Mandatory=$true, ParameterSetName="search")]
+            [ ValidateSet("contains", "ends_with", "starts_with", "equals", "notequals", "regex", IgnoreCase=$true) ]
+            [String]$criteria,
+        [Parameter(Mandatory=$true, ParameterSetName="search")]
+            [ ValidateNotNullOrEmpty() ]
+            [String]$value,
+        [Parameter(Mandatory=$true, ParameterSetName="entity")]
+        [Parameter(Mandatory=$true, ParameterSetName="search")]
+            [ ValidateSet("OR", "AND", IgnoreCase=$true) ]
+            [String]$operator,
+        [Parameter(Mandatory=$true, ParameterSetName="entity")]
+            [ ValidateNotNullOrEmpty() ]
+            [object]$entity
+    )
+
+    begin {
+
+        switch ( $criteria ) {
+
+            "equals" {
+                [string]$criteria = "="
+            }
+
+            "notequals" {
+                [string]$criteria = "!="
+            }
+
+            "regex" {
+                [string]$criteria = "similar_to"
+            }
+
+        }
+
+        #Populate the global membertype cache if not already done
+        #Using the API rather than hardcoding incase this changes with versions of NSX
+        if ( -not (test-path Variable:\NsxMemberTypes) ) {
+            $script:NsxMemberTypes = Get-NsxSecurityGroupMemberTypes
+        }
+
+        # TODO: [DC] Maybe in the future making the cmdlet aware of a number of
+        # entities being and creating the corresponding number of specs for it.
+        $entityCount = @($entity).count
+        if ( $entityCount -ne 1 ) {
+            throw "Multiple ($entityCount) entities specified . Only 1 is allowed."
+        }
+
+    }
+
+    process {
+
+        [System.XML.XMLDocument]$xmlDoc = New-Object System.XML.XMLDocument
+        [System.XML.XMLElement]$xmlDynamicCriteria = $XMLDoc.CreateElement("dynamicCriteria")
+        $xmlDoc.appendChild($xmlDynamicCriteria) | out-null
+
+        Add-XmlElement -xmlRoot $xmlDynamicCriteria -xmlElementName "operator" -xmlElementText $operator.ToUpper()
+
+        if ($PSCmdlet.ParameterSetName -eq "entity") {
+
+            # if ($_Member -is [System.Xml.XmlElement] ) {
+            if ($entity -is [System.Xml.XmlElement] ) {
+                $EntityObjectId = $entity.objectId
+            }
+            elseif ( ($entity -is [string]) -and ($entity -match "^vm-\d+$|^resgroup-\d+$|^dvportgroup-\d+$" )) {
+                $EntityObjectId = $entity
+            }
+            # TODO: [DC] As Nick what this is for?
```

Comment:
This was taken from the add security group member code. Technically it is valid (if your original code was) as anything you can add as a security group member can be used in the dynamic criteria "entity belongs to".